From 8f13941b6ef2f7c3bc75cd53d92cef6a87aeb6b4 Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Mon, 17 Dec 2018 22:58:30 +0800
Subject: [PATCH] DRVLIB_SW-846 - Changes in ltq eth driver to support aquantia
 PHY

- Remove phydev masking as it is not needed. We should rely
  on phy driver feature instead.
- Add checks for phy driver. In cases where phy driver is
  not found (e.g. if driver not enabled or firmware not up
  during bootup), phy framework will force configure it to
  generic phy driver, which does not work well for aquantia
  PHY (it makes the phy malfunction). So in this case we
  avoid the force reconfiguring and return fail instead.
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index bf704db661d6..5bde7a99a0d6 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -1764,22 +1764,19 @@ static int prx300_phy_connect(struct net_device *dev, struct xrx500_port *port)
 
 	priv = netdev_priv(dev);
 
+	phydev = of_phy_find_device(port->phy_node);
+	if (!phydev || !phydev->mdio.dev.driver) {
+		netdev_err(dev, "Unable to find phydev\n");
+		return -ENODEV;
+	}
+
 	phydev = of_phy_connect(dev, port->phy_node, &xrx500_mdio_link,
 				0, port->phy_if);
 	if (!phydev) {
-		netdev_err(dev, "Unable to find phydev\n");
+		netdev_err(dev, "Unable to connect phydev\n");
 		return -ENODEV;
 	}
 
-	phydev->supported &= (SUPPORTED_10baseT_Half
-			      | SUPPORTED_10baseT_Full
-			      | SUPPORTED_100baseT_Half
-			      | SUPPORTED_100baseT_Full
-			      | SUPPORTED_1000baseT_Half
-			      | SUPPORTED_1000baseT_Full
-			      | SUPPORTED_Autoneg
-			      | SUPPORTED_MII
-			      | SUPPORTED_TP);
 	phydev->advertising = phydev->supported;
 	port->phydev = phydev;
 
